### PR TITLE
Make current setting work for decays too

### PR DESCRIPTION
**Pull request by NNSSA**
The following things are added/changed:

- Parameter "kind" is included again. Downside of the environment variable "SPLIT_COLLISION_INTEGRAL" is that it must be changed each time one wants to do a different test. So the code makes a distinction now when kind = [0, 3] (the full integral) or kind = [1, 2, 4, 5]. For the latter to work, the environment variable must be set to False.

- Adams-Bashforth multistep method made the decay test fail, which is why I removed it. It shows very unstable behaviour.

- I removed the lower bound for decays, since it turned out that a small part of the allowed energy-momentum space is not included.

### CODE REVIEW

File: particles/__init__.py [hunk 1]
```diff
@@ -262,12 +262,9 @@ def calculate_collision_integral(self, ps):
             return numpy.zeros(len(ps))
 
         if not environment.get('SPLIT_COLLISION_INTEGRAL'):
-            fs = [sum([integral.integrate(ps, stepsize=self.params.h)
-                       for integral in self.collision_integrals])]
-
-            fs = list(self.data['collision_integral'][-MAX_ADAMS_BASHFORTH_ORDER:]) + fs
-
-            return adams_bashforth_correction(fs=fs, h=self.params.h) / self.params.h
```
Comment (ckald) [inline particles/__init__.py]: Why?

File: interactions/four_particle/__init__.py [hunk 2]
```diff
@@ -144,10 +144,14 @@ def integrate(self, ps, stepsize=None):
         if not self.cMs:
             self.cMs = [M_t(list(M.order), M.K1 / unit, M.K2 / unit) for M in self.Ms]
 
-        if environment.get('SPLIT_COLLISION_INTEGRAL'):
-            A = integration(ps, *bounds, self.creaction, self.cMs, stepsize, CollisionIntegralKind.F_1)
-            B = integration(ps, *bounds, self.creaction, self.cMs, stepsize, CollisionIntegralKind.F_f)
+        if environment.get('SPLIT_COLLISION_INTEGRAL') and self.kind in [0, 3]:
+            A = integration(ps, *bounds, self.creaction, self.cMs, stepsize, self.kind + 1)
+            B = integration(ps, *bounds, self.creaction, self.cMs, stepsize, self.kind + 2)
```
Comment (ckald) [inline interactions/four_particle/__init__.py]: You destroyed the whole idea of having constants. You shouldn't do algebra with these. Values are unimportant, they are arbitrary — only the variable names are important

File: tests/decay_lifetime_light/__main__.py [hunk 3]
```diff
@@ -76,12 +76,10 @@
...
-# 3: I_coll = A_vacuum_decay
-# 4: I_coll = f_1 * B_vacuum_decay
-# 5: I_coll = A_vacuum_decay + f_1 * B_vacuum_decay
-kind = 4
-
-# TODO: add variable that controls integrator precision
+# 3: I_coll = I_coll = A_vacuum_decay + f_1 * B_vacuum_decay
+# 4: I_coll = A_vacuum_decay
+# 5: I_coll = f_1 * B_vacuum_decay
+kind = 5
```
Comment (ckald) [inline tests/decay_lifetime_light/__main__.py]: No numeric values in the code!

File: tests/standard_model_bbn/__main__.py [hunk 4]
```diff
@@ -51,11 +51,20 @@
...
 ])
 
+# kind =
+# 0: I_coll = A + f_1 * B
+# 1: I_coll = A
+# 2: I_coll = f_1 * B
+# 3: I_coll = I_coll = A_vacuum_decay + f_1 * B_vacuum_decay
+# 4: I_coll = A_vacuum_decay
+# 5: I_coll = f_1 * B_vacuum_decay
+kind = 0
```
Comment (ckald) [inline tests/standard_model_bbn/__main__.py]: Same

File: tests/standard_model_bbn/__main__.py [hunk 5]
```diff
@@ -51,11 +51,20 @@
...
 
 universe.interactions += (
     SMI.neutrino_interactions(
         leptons=[electron],
-        neutrinos=[neutrino_e, neutrino_mu]
+        neutrinos=[neutrino_e, neutrino_mu],
+        kind=kind
```
Comment (ckald) [inline tests/standard_model_bbn/__main__.py]: This test does not need to know about existence of `kind` at all

Comment (ckald) [reply tests/standard_model_bbn/__main__.py]: We invented `kind` to switch on the vacuum decay approximation, no trace of it should be here

File: particles/__init__.py [hunk 6]
```diff
@@ -261,7 +262,7 @@ def calculate_collision_integral(self, ps):
         if not self.collision_integrals:
             return numpy.zeros(len(ps))
 
-        if not environment.get('SPLIT_COLLISION_INTEGRAL'):
+        if not environment.get('SPLIT_COLLISION_INTEGRAL') or os.environ.get('SPLIT_COLLISION_INTEGRAL') == 'False':
```
Comment (ckald) [inline particles/__init__.py]: Can we do without the second check?

Comment (NNSSA) [reply particles/__init__.py]: Couldn't get it work without...

File: interactions/four_particle/__init__.py [hunk 7]
```diff
@@ -144,10 +145,14 @@ def integrate(self, ps, stepsize=None):
         if not self.cMs:
             self.cMs = [M_t(list(M.order), M.K1 / unit, M.K2 / unit) for M in self.Ms]
 
-        if environment.get('SPLIT_COLLISION_INTEGRAL'):
```
Comment (ckald) [inline interactions/four_particle/__init__.py]: Like this

File: interactions/four_particle/__init__.py [hunk 8]
```diff
@@ -144,10 +145,14 @@ def integrate(self, ps, stepsize=None):
...
+        if environment.get('SPLIT_COLLISION_INTEGRAL') and not os.environ.get('SPLIT_COLLISION_INTEGRAL') == 'False':
```
Comment (ckald) [inline interactions/four_particle/__init__.py]: And not like this

**Pull request merged**